addons/account_edi_proxy_client/models/account_edi_proxy_user.py

```diff
@@ -0,0 +1,158 @@
+    edi_format_code = fields.Char(related='edi_format_id.code', readonly=True)
+    edi_identification = fields.Char(required=True, help="The unique id that identifies this user for on the edi format, typically the vat")
+    private_key = fields.Binary(required=True, attachment=False, groups="base.group_system", help="The key to encrypt all the user's data")
+    refresh_token = fields.Char(groups="base.group_system")
+
+    _sql_constraints = [
+        ('unique_id_client', 'unique(id_client)', 'This id_client is already used on another user.'),
+        ('unique_edi_identification_per_format', 'unique(edi_identification, edi_format_id)', 'This edi identification is already assigned to a user'),
+    ]
+
+    def _make_request(self, url, params):
+        ''' Make a request to proxy and handle the generic elements of the reponse (errors, new refresh token).
+        '''
+        payload = {
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'params': params,
+            'id': uuid.uuid4().hex,
+        }
+
+        try:
+            response = requests.post(
+                url,
+                json=payload,
+                timeout=TIMEOUT,
+                headers={'content-type': 'application/json'},
+                auth=OdooEdiProxyAuth(user=self)).json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError):
+            raise AccountEdiProxyError('connection_error',
+                _('The url that this service requested returned an error. The url it tried to contact was %s', url))
+
+        proxy_error = response['result'].pop('proxy_error', False)
+        if proxy_error:
+            error_code = proxy_error['code']
+            if error_code == 'refresh_token_expired':
+                self.renew_token()
+                return self._make_request(url, params)
+            if error_code == 'no_such_user':
+                # This error is also raised if the user didn't exchange data and someone else claimed the edi_identificaiton.
+                self.active = False
+            raise AccountEdiProxyError(error_code, proxy_error['message'] or False)
+
+        return response['result']
+
+    def _register_proxy_user(self, company, edi_format, edi_identification):
+        ''' Generate the public_key/private_key that will be used to encrypt the file, send a request to the proxy
+        to register the user with the public key and create the user with the private key.
+
+        :param company: the company of the user.
+        :param edi_identification: The unique ID that identifies this user on this edi network and to which the files will be addressed.
+                                   Typically the vat.
+        '''
+        private_key = rsa.generate_private_key(
+            public_exponent=65537,
+            key_size=2048,
```

Comment:
This is the security between the client and the proxy so ours to choose.

Reply:
Ok, these are default values suggested by the documentation of 'cryptography'. I read some stuff about it and saw no reason to change it. But as it is not my area of expertise, if you have a better suggestion, I'm open to change that.
I added some comments to make it clearer in the code.

I also added some general comments about the flows that are supported here. I had already done that following Julien's comment but I agree it was still lacking a bit on this side. Please note that if you really want to understand all the flows of this PR, you would also need the documentation of 'l10n_it_edi' and 'account_edi' which can be quite of a steep curve :D Hopefully this is not needed for the security review since most of the 'security sensitive' stuff (appart from requests) is made in 'account_edi_proxy_client'.